packages/account/src/Sections/Profile/LanguageSettings/language-settings.tsx

```diff
@@ -2,29 +2,24 @@ import React from 'react';
 import { useTranslation } from 'react-i18next';
 import { Button, DesktopWrapper } from '@deriv/components';
 import { localize, getAllowedLanguages } from '@deriv/translations';
-import { connect } from 'Stores/connect';
+import { observer, useStore } from '@deriv/stores';
 import FormSubHeader from 'Components/form-sub-header';
-import TCoreStore from '../../../Stores';
 import { Formik, FormikHandlers, FormikHelpers, FormikValues } from 'formik';
 import FormFooter from 'Components/form-footer';
 import LanguageRadioButton from 'Components/language-settings';
 
-type TLanguageSettings = {
-    current_language: string;
-    changeSelectedLanguage: (lang: string) => void;
-    isCurrentLanguage: (lang: string) => boolean;
-};
-
-const LanguageSettings = ({ changeSelectedLanguage, current_language, isCurrentLanguage }: TLanguageSettings) => {
+const LanguageSettings = observer(() => {
     const { i18n } = useTranslation();
+    const { common } = useStore();
+    const { changeLanguage, changeCurrentLanguage, current_language, isCurrentLanguage } = common;
     const allowed_language_keys: string[] = Object.keys(getAllowedLanguages());
     const initial_values = { language_code: current_language };
     return (
         <Formik
             initialValues={initial_values}
-            onSubmit={async values => {
+            onSubmit={async (values: { language_code: any }) => {
                 const { language_code } = values;
-                changeSelectedLanguage(language_code);
+                await changeLanguage(language_code, changeCurrentLanguage);
```

Comment:
what are these changes? we don’t have changeLanguage in common-store, did I miss something?